### ISSUE TITLE
Inline type annotations have wrong precedence

### ISSUE THREAD

**Issue by natefaubion** [purescript/purescript#3554]
Found this in the wild:
```purescript
foldMap1 f = foldMap1 f <<< toUnfoldable1 :: forall a. NonEmptySet a -> NonEmptyList a
```

`::` Currently binds tighter than operators. It should have lowest precedence.

**Comment by hdgarrood** [comment]
Fixed by #3555.

**Comment by hdgarrood** [comment]
although I suppose we could still do with a test
